bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java

```diff
@@ -3,41 +3,79 @@
 // Licensed under the Apache License, Version 2.0.
 package org.ethereum.evmc;
 
-import org.ethereum.evmc.EvmcLoaderException;
-
+import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.util.Objects;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
 
 /**
  * The Java interface to the evm instance.
  *
  * <p>Defines the Java methods capable of accessing the evm implementation.
  */
 public final class EvmcVm implements AutoCloseable {
-  private static EvmcVm evmcVm;
-  private static boolean isEvmcLibraryLoaded = false;
+  private static final Throwable errorLoadingEvmc;
   private ByteBuffer nativeVm;
+
+  static {
+    Throwable error = null;
+    try {
+      // load so containing the jni bindings to evmc
+      System.loadLibrary("libevmc-java");
+    } catch (UnsatisfiedLinkError e) {
+      String extension = null;
+      String os = System.getProperty("os.name").toLowerCase();
+      if (os.contains("win")) {
+        extension = "dll";
+      } else if (os.contains("nix") || os.contains("nux") || os.contains("aix")) {
+        extension = "so";
+      } else if (os.contains("mac")) {
+        extension = "dylib";
+      } else {
+        error = e;
+      }
+      if (extension != null) {
+        try {
+          Path evmcLib = Files.createTempFile("libevmc-java", extension);
+          Files.copy(
+              EvmcVm.class.getResourceAsStream("/libevmc-java." + extension),
+              evmcLib,
+              StandardCopyOption.REPLACE_EXISTING);
+          evmcLib.toFile().deleteOnExit();
+          try {
+            System.load(evmcLib.toAbsolutePath().toString());
+          } catch (UnsatisfiedLinkError e1) {
+            error = e1;
+          }
+        } catch (IOException ex) {
+          error = ex;
+        }
+      }
+    }
+    errorLoadingEvmc = error;
+  }
+
+  /**
+   * Returns true if the native library was loaded successfully and EVMC capabilities are available.
+   *
+   * @return true if the library is available
+   */
+  public static boolean isAvailable() {
```

Comment:
Must this be public? Seems to be only used inside `create()`. Drop it if not needed.

Reply:
Yes, otherwise another program located in a different java package cannot check if EVMC is available.